### PR TITLE
Shelllcheck fixes for shell scripts in ./hack directory

### PR DESCRIPTION
**Pull request by achinthagunasekara**
Shelllcheck fixes for shell scripts in ./hack directory

### CODE REVIEW

File: hack/for-go-proj.sh [hunk 1]
```diff
@@ -49,6 +49,7 @@ for godep_file in ${godep_projects}; do
     cd "${project}"
     case "${CMD}" in
       "test")
+        #shellcheck disable=SC2046
```
Comment (MaciekPytel) [inline hack/for-go-proj.sh]: Can we avoid adding all those comments everywhere?

File: hack/for-go-proj.sh [hunk 2]
```diff
@@ -18,7 +18,7 @@ set -o errexit
 set -o pipefail
 set -o nounset
 
-CONTRIB_ROOT="$(dirname ${BASH_SOURCE})/.."
+CONTRIB_ROOT="$(dirname "${BASH_SOURCE[0]}")/.."
```
Comment (kgolab) [inline hack/for-go-proj.sh]: I'm not sure that the extra quotation marks here help any, the whole variable is already surrounded by quotes.
Similar issue is visible in kube-env.sh.

File: hack/install-verify-tools.sh [hunk 3]
```diff
@@ -18,7 +18,8 @@ set -o errexit
 set -o nounset
 set -o pipefail
 
-KUBE_ROOT=$(dirname "${BASH_SOURCE}")/..
+KUBE_ROOT=$(dirname "${BASH_SOURCE[0]}")/..
+"${KUBE_ROOT}"/hack/check-go-version.sh
```
Comment (kgolab) [inline hack/install-verify-tools.sh]: I think it would be good not to mix functional changes with formatting/code fixes.

**Comment by MaciekPytel** [comment]
Have you actually tested all those scripts after the changes? I'm suspicious about just applying linter changes to shell scripts, it's just so easy for those to have unexpected behavior changes (especially when adding quotes to bash commands, which this PR seems to do quite liberally).

**Comment by achinthagunasekara** [comment]
I'll break this into smaller PRs and do more testing :)

**Pull request closed without merging**